File: sub-module-web-service/src/main/java/org/github/web/module/sys/menu/mapper/IMenuMapper.java
package org.github.web.module.sys.menu.mapper;

import lombok.*;
import org.github.base.IPage;
import org.github.mybatis.MyBatisMapper;
import org.github.web.module.sys.menu.dto.QueryMenuListBO;
import org.github.web.module.sys.menu.dto.QueryMenuListVO;
import org.apache.ibatis.annotations.Param;

@MyBatisMapper
public interface IMenuMapper {
  @NonNull IPage<QueryMenuListVO> queryPage(@Param("page") IPage<QueryMenuListVO> page, @Param("bo") QueryMenuListBO bo);
}

File: sub-module-web-service/src/main/java/org/github/web/module/sys/menu/mapper/IMenuMapper.kt
package org.github.web.module.sys.menu.mapper

import org.github.base.IPage
import org.github.mybatis.MyBatisMapper
import org.github.web.module.sys.menu.model.vo.QueryMenuListVO
import org.github.web.module.sys.menu.model.bo.QueryMenuListBO

@MyBatisMapper
interface IMenuMapper {
  fun queryPage(page: IPage<QueryMenuListVO>, bo: QueryMenuListBO): IPage<QueryMenuListVO>
}
